youtube_dl/extractor/funimation.py

```diff
@@ -87,26 +87,40 @@ def _login(self):
         (username, password) = self._get_login_info()
         if username is None:
             return
+
         data = urlencode_postdata({
             'email_field': username,
             'password_field': password,
         })
         user_agent = self._extract_cloudflare_session_ua(self._LOGIN_URL)
         if not user_agent:
             user_agent = 'Mozilla/5.0 (Windows NT 5.2; WOW64; rv:42.0) Gecko/20100101 Firefox/42.0'
+
+        # Request home page to establish initial session and cookies
+        initial_request = sanitized_Request(self._LOGIN_URL, headers={
+            'User-Agent': user_agent,
+            'Content-Type': 'application/x-www-form-urlencoded'
+        })
+        self._download_webpage(initial_request, None, 'Starting session with Funimation')
```

Comment:
`headers` now can be passed right in `_download_webpage`. Also there is already the same dict in `login_request` that may be extracted and reused.